fasttrips/Trip.py

```diff
@@ -343,44 +304,43 @@ def __init__(self, input_dir, output_dir, gtfs_schedule, today, stops, routes, p
         FastTripsLogger.debug("Final (%d)\n%s" % (len(self.trips_df), str(self.trips_df.head())))
         FastTripsLogger.debug("\n"+str(self.trips_df.dtypes))
 
-        self.stop_times_df = pandas.DataFrame(data=stop_time_dicts)
+        self.stop_times_df = gtfs_feed.stop_times
 
         # Read the fast-trips supplemental stop times data file
-        if os.path.exists(os.path.join(input_dir, Trip.INPUT_STOPTIMES_FILE)):
-            stop_times_ft_df = pandas.read_csv(os.path.join(input_dir, Trip.INPUT_STOPTIMES_FILE),
-                                               skipinitialspace=True,
-                                          dtype={Trip.STOPTIMES_COLUMN_TRIP_ID:object,
-                                                 Trip.STOPTIMES_COLUMN_STOP_ID:object})
-            # verify required columns are present
-            stop_times_ft_cols = list(stop_times_ft_df)
-            assert({Trip.STOPTIMES_COLUMN_TRIP_ID, Trip.STOPTIMES_COLUMN_STOP_ID}.issubset(stop_times_ft_df))
-
-            # Join to the trips dataframe
-            if len(stop_times_ft_df.columns) > 2:
-                self.stop_times_df = pandas.merge(left=self.stop_times_df, right=stop_times_ft_df,
-                                                  how='left',
-                                                  on=[Trip.STOPTIMES_COLUMN_TRIP_ID,
-                                                      Trip.STOPTIMES_COLUMN_STOP_ID])
+        with zipfile.ZipFile(input_archive, 'r') as zipf:
+            if Trip.INPUT_STOPTIMES_FILE in zipf.namelist():
+                stop_times_ft_df = pandas.read_csv(zipf.open(Trip.INPUT_STOPTIMES_FILE),
```

Comment:
are these not already in a DF from reading them into partridge?

Reply:
These lines of code are reading in the stop_times_ft.txt. The stop_times.txt is read in directly as a pandas DF via Partridge. Ideally, Partridge would be agnostic and read in everything in the GTFS zip, but that seems like kind of big deal. I'm not sure of the impact on their side of the fence. Let me know if I'm missing the point of your question.

Reply:
You can read these files with Partridge and the default type conversions treat everything as a string. This is part of the public API of Partridge.

```python
feed = ptg.feed('path/to/inputs/')
df = feed.get('stop_times_ft.txt')

# then do your manual type coercions
```

To override default type coercions:

```python
config = ptg.config.default_config()
config.add_node('stop_times_ft.txt', {
    'converters': {
        'arrival_time': my_vectorized_parse_function,
    },
})

feed = ptg.feed('path/to/inputs/', config=config)
feed.get('stop_times_ft.txt')
```

You can also describe the relationships between files as edges of the config graph:

```python
config = ptg.config.default_config()
config.add_node('stop_times_ft.txt')
config.add_edge('stop_times_ft.txt', 'trips.txt', {
    'dependencies': {
        'trip_id': 'trip_id',
    },
})

feed = ptg.feed('path/to/inputs/', config=config)
feed.get('stop_times_ft.txt')
```

The second and third approaches are not considered to be part of the public API, but I would like them to be someday. So I can't guarantee that the expected node/edge data formats will remain the same in future releases.

Let me know if I can clarify anything.

Reply:
See Partidge’s config.py file for more context.

Reply:
This is pretty neat stuff. I implemented option 1, and I think I will go through and implement at least option 2 for type casting. Option 3 would be really cool and it would save a lot of code in Fast-Trips.